### PR TITLE
fix leaking of pool.Message

### PR DESCRIPTION
**Pull request by jkralik**

### CODE REVIEW

File: net/blockwise/blockwise_test.go [hunk 1]
```diff
@@ -303,15 +304,20 @@ func TestBlockWiseDo(t *testing.T) {
...
+			require.Len(t, receivingMessagesCache, 0)
+			sendingMessagesCache := sender.sendingMessagesCache.PullOutAll()
+			require.Len(t, sendingMessagesCache, 0)
 			require.NoError(t, err)
 			assert.Equal(t, tt.want, fromPoolMessage(got))
 		})
 	}
 }
 
+/*
```
Comment (Danielius1922) [inline net/blockwise/blockwise_test.go:317]: The parallel test doesn't work anymore?

Comment (jkralik) [reply net/blockwise/blockwise_test.go:317]: No this need to be fixed separately.

File: net/blockwise/blockwise.go [hunk 2]
```diff
@@ -429,6 +380,23 @@ func (b *BlockWise[C]) sendEntityIncomplete(w *responsewriter.ResponseWriter[C],
 	w.SetMessage(sendMessage)
 }
 
+func wantToReceived(r *pool.Message) bool {
```
Comment (Danielius1922) [inline net/blockwise/blockwise.go:383]: [nitpick - bad grammar] `wantsToReceive` or `wantsToBeReceived`

File: net/blockwise/blockwise.go [hunk 3]
```diff
@@ -462,8 +430,8 @@ func (b *BlockWise[C]) Handle(w *responsewriter.ResponseWriter[C], r *pool.Messa
 		b.errors(fmt.Errorf("continueSendingMessage(%v): %w", r, err))
 		return
 	}
-	if b.autoCleanUpResponseCache && !more {
-		b.RemoveFromResponseCache(token)
+	if !more && sendingMessageCached.Data().Code() > codes.DELETE {
```
Comment (Danielius1922) [inline net/blockwise/blockwise.go:433]: Are codes `>` than `codes.DELETE` invalid or why are only those codes deleted from cache?

**Review by Danielius1922 — APPROVED** [review]

File: net/blockwise/blockwise.go (hunk 3 above)
Comment (jkralik) [reply net/blockwise/blockwise.go:426]: For codes GET,POST,PUT,DELETE, we want them to wait for pairing response and then delete them when the full response comes in or when timeout occurs.

File: net/blockwise/blockwise.go (hunk 2 above)
Comment (jkralik) [reply net/blockwise/blockwise.go]: fixed

**Pull request merged**